src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarm.java

```diff
@@ -488,7 +488,35 @@ public ListSuitesResult listSuites(String runArn) {
         return api.listSuites(request);
     }
 
+    public int getUnmeteredDevices(String os) {
+        AccountSettings accountSettings = getAccountSettings();
+        if (accountSettings == null) {
+            return 0;
+        } else if (os.equalsIgnoreCase("ANDROID")) {
+            return getAccountSettings().getUnmeteredDevices().get("ANDROID");
+        } else if (os.equalsIgnoreCase("IOS")) {
+            return getAccountSettings().getUnmeteredDevices().get("IOS");
+        } else {
+            return 0;
+        }
+    }
 
+    public String getOs(String appArtifact) {
+        if (appArtifact.toLowerCase().endsWith("apk")) {
+            return "Android";
+        } else {
```

Comment:
Because this code gets invoked on form validation, jenkins cannot recognize the exception . I agree we would default the user to IOS with a  ZIP. But when actual build will run, we validate again and throw a exception at that time